### ISSUE TITLE
DateField doesn't propogate onTimeChange

### ISSUE THREAD

**Issue by cpietrzykowski** [zippyui/react-date-picker#94]
When including the ClockInput component the only way for the field to update on a time change is using the Ok button -- it appears. The code for [onTimeChange ](https://github.com/zippyui/react-date-picker/blob/fe0a24b091f37a752efb194eedebbc48b4d8462c/src/DateField/index.js#L411-L420)(in DateField) only sets the time property in the object, and as such the caller has no way to know when the time is changed.

If I'm in error I would appreciate advise. A patch seems simple enough, unless there are objections I would consider a patch if I get a bit more familiar with how the components work together.

**Comment by radubrehar** [comment]
The DateField is only supposed to let the user know when a change occurs, be it time and/or date change.

```jsx
const onChange = (dateString, { dateMoment, timestamp }) => {
  ....
}
<DateField onChange={onChange} />
```
On the `onChange` prop, you have both the date/time both as a string and as a moment.js instance, you if you are insterested on time updates, you can compare the current value with the previous value.
Does this help? Or can you describe what exactly you're trying to achieve?

**Comment by cpietrzykowski** [comment]
Here's the setup I had in mind, you tell me if I'm out to lunch so to speak.

User clicks DateField (showClock, updateOnDateClick, footer: false) with a default of now. User moves time ahead a couple hours. DateField loses focus and collapses. Time input is lost.

I would have expected an onChange event there, at least on focus being lost. It might be my misunderstanding of your design and getting a ux clash with updateOnDateClick enabled.

**Comment by pointblack** [comment]
I have the issue that the change event is fired,  the timestamp is always 00:00:00, even when i change the time. dateString is passing the correct date value

Component:

```
<Calendar
  dateFormat="YYYY-MM-DD HH:mm"
  date={deliveryDate}
  onChange={onDeliveryDateChange}
/>
```

Container
```
    this.onDeliveryDateChange = (dateString, { dateMoment, timestamp }) => {
        console.log(timestamp); // Wed Jul 06 2016 00:00:00 GMT+0200
        console.log(dateMoment);
    };
```

**Comment by radubrehar** [comment]
Just fixed this in `5.3.22` - also reported in issue #122
